autosklearn/ensembles/ensemble_selection.py

```diff
@@ -64,38 +62,59 @@ def _fast(self, predictions, labels):
 
         ensemble_size = self.ensemble_size
 
-        if self.sorted_initialization:
-            n_best = 20
-            indices = self._sorted_initialization(predictions, labels, n_best)
-            for idx in indices:
-                ensemble.append(predictions[idx])
-                order.append(idx)
-                ensemble_ = np.array(ensemble).mean(axis=0)
-                ensemble_performance = calculate_score(
-                    labels, ensemble_, self.task_type, self.metric,
-                    ensemble_.shape[1])
-                trajectory.append(ensemble_performance)
-            ensemble_size -= n_best
-
+        weighted_ensemble_prediction = np.zeros(predictions[0].shape)
+        fant_ensemble_prediction = np.zeros(weighted_ensemble_prediction.shape)
         for i in range(ensemble_size):
             scores = np.zeros((len(predictions)))
             s = len(ensemble)
             if s == 0:
-                weighted_ensemble_prediction = np.zeros(predictions[0].shape)
+                np.multiply(
+                    weighted_ensemble_prediction,
+                    0,
+                    out=weighted_ensemble_prediction,
+                )
             else:
                 # Memory-efficient averaging!
-                ensemble_prediction = np.zeros(ensemble[0].shape)
+                np.multiply(
+                    weighted_ensemble_prediction,
+                    0,
+                    out=weighted_ensemble_prediction,
+                )
                 for pred in ensemble:
-                    ensemble_prediction += pred
-                ensemble_prediction /= s
-
-                weighted_ensemble_prediction = (s / float(s + 1)) * ensemble_prediction
-            fant_ensemble_prediction = np.zeros(weighted_ensemble_prediction.shape)
+                    np.add(
+                        weighted_ensemble_prediction,
+                        pred,
+                        out=weighted_ensemble_prediction,
+                    )
+                np.multiply(
+                    weighted_ensemble_prediction,
+                    1/s,
+                    out=weighted_ensemble_prediction,
+                )
+
+                np.multiply(
+                    weighted_ensemble_prediction,
+                    (s / float(s + 1)),
+                    out=weighted_ensemble_prediction,
+                )
+            np.multiply(
+                fant_ensemble_prediction,
+                0,
+                out=fant_ensemble_prediction,
+            )
             for j, pred in enumerate(predictions):
                 # TODO: this could potentially be vectorized! - let's profile
                 # the script first!
-                fant_ensemble_prediction[:, :] = \
-                    weighted_ensemble_prediction + (1. / float(s + 1)) * pred
+                np.add(
```

Comment:
This will not clear the predictions, but in each round add them to the predictions of the round before. Could you please continue using the [:, :] operation as this does a copy of the data into the array?

Reply:
This could also make the fill above irrelevant

Reply:
added np.fill/np.add combination instead, as previous + operator increased the memory by creating extra arrays